### PR TITLE
fix(replication): ignore the problem of not getting Content-Length fr…

### PR DESCRIPTION
**Pull request by chlins**
…om the pull blob api

Fix: https://github.com/goharbor/harbor/issues/11858, https://github.com/goharbor/harbor/issues/9216
Signed-off-by: chlins <chlins.zhang@gmail.com>

### CODE REVIEW

**Comment by reasonerjt** [comment]
We must make sure this doesn't impact the correctness for quota usage.

File: src/replication/transfer/image/transfer.go [hunk 1]
```diff
@@ -269,6 +272,14 @@ func (t *transfer) copyBlob(srcRepo, dstRepo, digest string) error {
 		return err
 	}
 	defer data.Close()
+
+	t.logger.Debugf("the blob size from api is %d, from manifests is %d", size, sizeFromDescriptor)
```
Comment (bitsf) [inline src/replication/transfer/image/transfer.go:276]: dup log as below

File: src/replication/transfer/image/transfer.go [hunk 2]
```diff
@@ -269,6 +272,14 @@ func (t *transfer) copyBlob(srcRepo, dstRepo, digest string) error {
...
+	// get size 0 from PullBlob, use size from distribution.Descriptor instead.
+	if size == 0 {
+		size = sizeFromDescriptor
+		t.logger.Debugf("the blob size from api is 0, use size %d from manifests instead", size)
```
Comment (heww) [inline src/replication/transfer/image/transfer.go:280]: `the blob size from api is 0, use size %d from manifests instead` to `the blob size from remote registry is 0, use size %d from manifests instead`

File: src/replication/transfer/image/transfer.go [hunk 3]
```diff
@@ -243,12 +245,13 @@ func (t *transfer) copyContent(content distribution.Descriptor, srcRepo, dstRepo
 	// the media type of the layer or config can be "application/octet-stream",
 	// schema1.MediaTypeManifestLayer, schema2.MediaTypeLayer, schema2.MediaTypeImageConfig
 	default:
-		return t.copyBlob(srcRepo, dstRepo, digest)
+		return t.copyBlob(srcRepo, dstRepo, digest, size)
```
Comment (heww) [inline src/replication/transfer/image/transfer.go:248]: `t.copyBlob(srcRepo, dstRepo, digest, size)` to `t.copyBlob(srcRepo, dstRepo, digest, conten.size)`

**Pull request merged**